src/Shared/PackageManagers/PyPIProjectManager.cs

```diff
@@ -467,7 +467,6 @@ public override Uri GetPackageAbsoluteUri(PackageURL purl)
             JsonDocument contentJSON = JsonDocument.Parse(metadata);
 
             List<string> possibleProperties = new() { "homepage", "home_page" };
-            List<string> searchPropertiesProjectURLs = new() { "Source", "Source Code", "homepage", "Bug Tracker", "Repository", "Bug Reports" };
```

Comment:
Per your report, for the repositories this isn't working for, the git url is held in the "download" property. It seems like the simpler solution would be to add "download" to the list of properties we search for. This seems like it might have some unintended side effects if there happen to be unrelated git repositories under a property we aren't anticipating.  Is there a reason why that wouldn't work?

Reply:
According to the reference of [`setup`](https://setuptools.pypa.io/en/latest/references/keywords.html), the `project_urls` filed is
> An arbitrary map of URL names to hyperlinks, allowing more extensible documentation of where various resources can be found than the simple url and download_url options provide.

So I think `project_urls` may contain code repository information in properties with arbitrary keys, such as Code, Issue Tracker, Documentation, etc. An example is `aws-satella` package. A conservative way is to maintain a list containing very possible keys. An aggressive way is to search all properties in the `project_urls` field.